src/hotspot/share/gc/g1/g1HeapRegion.hpp

```diff
@@ -74,7 +74,7 @@ class HeapRegion : public CHeapObj<mtGC> {
 
   HeapWord* volatile _top;
 
-  G1BlockOffsetTablePart _bot_part;
+  G1BlockOffsetTable* _bot;
```

Comment:
I suppose there is no longer a reason for this to be part of the region. Any downside to referring to it using `g1h->bot()`?

Reply:
That would require calling `G1CollectedHeap::heap()` every time bot is needed. Actually, maybe some methods, e.g. `update_bot_for_block` should belong to heap, instead of heap-region.

(Either way, I feel this decision can/should be made in its own ticket.)

Reply:
> (Either way, I feel this decision can/should be made in its own ticket.)

I agree. This patch should focus on merging.